src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/LincheckJavaAgent.kt

```diff
@@ -133,8 +133,10 @@ internal object LincheckJavaAgent {
                     // old classes that were already loaded before and have coroutine method calls inside
                     canonicalClassName in coroutineCallingClasses
                 }
-                instrumentation.retransformClasses(*classes.toTypedArray())
-                instrumentedClasses.addAll(classes.map { it.name })
+                if (classes.isNotEmpty()) {
+                    instrumentation.retransformClasses(*classes.toTypedArray())
```

Comment:
nit: `retransformClasses` on empty array is [guaranteed](https://docs.oracle.com/javase/8/docs/api/java/lang/instrument/Instrumentation.html#:~:text=retransformClasses,-void%20retransformClasses(Class&text=classes)%20throws%20UnmodifiableClassException-,Retransform%20the%20supplied%20set%20of%20classes.,be%20transformed%20with%20the%20ClassFileTransformer%20.) no-op, so this `if` is exhaustive